### PR TITLE
Draft release notes page for CVL 2

### PR DESCRIPTION
**Pull request by mdgeorge4153**
This is based on the internal changelog in confluence

Generated: https://certora-certora-prover-documentation--104.com.readthedocs.build/en/104/docs/prover/changelog.html

### CODE REVIEW

File: docs/cvl/expr.md [hunk 1]
```diff
@@ -359,3 +359,11 @@ state `s`.
 Unresolved method calls
 ```
 
+(storage-comparison)=
+Comparing storage
+-----------------
+
+```{todo}
+Merge #78
```
Comment (shellygr) [inline docs/cvl/expr.md:367]: I don't think we want that to reach public docs

File: docs/prover/changelog.md [hunk 2]
```diff
@@ -1,517 +1,79 @@
...
-Prod version November 16, 2021
-------------------------------
-
-*   Fix to how remappings are set
-    
-*   Fixing bugs in static analyses relevant when handling structs in optimized solidity code
-    
-*   Fixing multiplication axioms
-    
+4.0.4 CVL 2 preview (beta)
```
Comment (shellygr) [inline docs/prover/changelog.md:19]: no need to include

File: docs/prover/changelog.md [hunk 3]
```diff
@@ -1,517 +1,79 @@
...
-    
-*   Fix to allow havoc assuming in non-hook context
-    
-*   Improved `calldata` splitting
-    
-*   More efficient axiom instantiation
-    
-*   Improved internal function finder
-    
+ - {ref}`storage-comparison`
```
Comment (shellygr) [inline docs/prover/changelog.md:24]: is it even in 4.0.5? I'm not sure

File: docs/prover/changelog.md [hunk 4]
```diff
@@ -1,517 +1,79 @@
...
 
-Prod version November 7, 2021
------------------------------
+```{todo}
+Needs more explanation / documentation:
+ - Support try-catch
```
Comment (shellygr) [inline docs/prover/changelog.md:28]: no need to mention

**Review by shellygr — COMMENTED** [review]
the diff is a bit challenging but added comments on how I think it should look

File: docs/cvl/expr.md (hunk 1 above)
Comment (mdgeorge4153) [reply docs/cvl/expr.md:367]: This is docs PR #78 - should be replaced with a link to the update documentation (which has landed)

File: docs/prover/changelog.md [hunk 5]
```diff
@@ -1,517 +1,79 @@
...
-    
-*   Making loop unroller more robust in optimized loops
-    
-*   Fixing addition optimization
-    
+ - Improved error messages
+ - Bug fixes and internal improvements
 
-Prod version July 18, 2021
+4.0.0 CVL 2 preview (beta)
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/prover/changelog.md:53]: I was thinking that when there's a new minor version, we consolidate all of the patch version changelogs into one entry.

File: docs/prover/changelog.md [hunk 6]
```diff
@@ -1,517 +1,79 @@
...
-*   Fix to program splitting
-    
-*   Bug fixes to Points-to analysis
-    
-*   Bug fix to CVL variables not appearing in call trace
-    
+```{todo}
+Needs documentation:
+  - add support for Vyper
+  - Support `CONSTANT` summaries for all non-dynamic return types
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/prover/changelog.md]: I mean, we need to update the existing document to describe the change, no? (at least for the second one)

[the remark this answers is not in the document]
Comment (shellygr) [reply docs/prover/changelog.md]: It's not that often used. The question is whether the existing docs say certain types are limited or not

File: docs/prover/changelog.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (shellygr) [reply docs/prover/changelog.md:45]: We haven't released anything before 4.0.5, so the contents of previous minors we had doesn't matter. In the future we want to, as you mentioned, consolidate all updates to the released version. 
So for example, if we have in beta 4.1.2, 4.1.3, 4.1.4 and then 4.1.4 becomes non-beta - we put the updates of 4.1.2, 4.1.3 and 4.1.4. (Internally we'll keep the full breakdown for the different beta versions)

File: docs/prover/changelog.md [hunk 7]
```diff
@@ -1,517 +1,79 @@
...
-    
-*   Support ALL (forced) external summaries and make this default: now behavior is consistent with internal summaries
-    
-*   Recent jobs file is now renamed when corrupted (or has an incompatible format).
-    
-*   Improvements to call trace
-    
+```{todo}
+Needs documentation:
+ - `--protocol_name` and `--protocol_owner`
```
[the remark this answers is not in the document]
Comment (shellygr) [reply docs/prover/changelog.md:56]: those options are automatically set by the CLI if not given manually, and I'm not sure the manual setting story makes sense now

File: docs/prover/changelog.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/prover/changelog.md]: makes sense

**Review by shellygr — APPROVED** [review]

**Pull request merged**